### PR TITLE
[CARBONDATA-3216] Fix enableLocalDictionary with false issue in CSDK

### PR DESCRIPTION
**Pull request by xubo245**
1.enableLocalDictionary can' t set false

Be sure to do all of the following checklist to help us incorporate 
your contribution quickly and easily:

 - [x] Any interfaces changed?
 No
 - [x] Any backward compatibility impacted?
 NO
 - [x] Document update required?
No
 - [x] Testing done
    No
 - [x] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA. 
JIRA2951

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/2074/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/2279/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.2, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/10328/

**Comment by xubo245** [comment]
@KanakaKumar @ajantha-bhat @jackylk @ravipesala Please review it.

File: store/CSDK/test/main.cpp [hunk 1]
```diff
@@ -546,7 +546,7 @@ bool testWriteData(JNIEnv *env, char *path, int argc, char *argv[]) {
         writer.withCsvInput(jsonSchema);
         writer.withLoadOption("complex_delimiter_level_1", "#");
         writer.writtenBy("CSDK");
-        writer.taskNo(185);
+        writer.taskNo(15541554.81);
```
Comment (KanakaKumar) [inline store/CSDK/test/main.cpp]: This change not related to fix

File: store/CSDK/test/main.cpp [hunk 2]
```diff
@@ -709,6 +709,7 @@ bool testWithTableProperty(JNIEnv *env, char *path, int argc, char **argv) {
         writer.outputPath(path);
         writer.withCsvInput(jsonSchema);
         writer.withTableProperty("sort_columns", "shortField");
+        writer.enableLocalDictionary(false);
```
Comment (KanakaKumar) [inline store/CSDK/test/main.cpp]: Can you test if there is any possibility to give NULL in C and how it is sent to java layer ? true or false?

File: store/CSDK/src/CarbonWriter.cpp [hunk 3]
```diff
@@ -291,9 +291,6 @@ void CarbonWriter::localDictionaryThreshold(int localDictionaryThreshold) {
 }
 
 void CarbonWriter::enableLocalDictionary(bool enableLocalDictionary) {
-    if (enableLocalDictionary == NULL) {
```
[the remark this answers is not in the document]
Comment (ajantha-bhat) [reply store/CSDK/src/CarbonWriter.cpp]: @KanakaKumar : In CPP, boolean false value is same as NULL. Both are zero. Hence when we pass false, it always throws exception due to this wrong code.  Hence he removed.

@xubo245 : This is just two removal, no need separate PR. you can do with #3048

File: store/CSDK/test/main.cpp [hunk 4]
```diff
@@ -853,7 +854,7 @@ int main(int argc, char *argv[]) {
     } else {
         int batch = 32000;
         int printNum = 32000;
-
+//
```
[the remark this answers is not in the document]
Comment (xubo245) [reply store/CSDK/test/main.cpp]: ok, done

File: store/CSDK/test/main.cpp (hunk 1 above)
Comment (xubo245) [reply store/CSDK/test/main.cpp]: There are some different  between CPP and java: java taskNo don't support double, but CPP will convert double to long.

File: store/CSDK/test/main.cpp (hunk 2 above)
Comment (xubo245) [reply store/CSDK/test/main.cpp]: add test code for it. In CPP, it will convert NULL to false when user invoke CarbonWriter::enableLocalDictionary.

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/2359/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.2, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/10402/

**Comment by xubo245** [comment]
@ajantha-bhat @KanakaKumar Please review it.

**Pull request closed without merging**